File: service/fe-web/src/utils/guid.ts
const SYSTEM_CODE = 'LTP';

export function generateGuid(): string {
  const now = new Date();
  const pad = (n: number, len: number) => String(n).padStart(len, '0');
  const timestamp =
    now.getFullYear().toString() +
    pad(now.getMonth() + 1, 2) +
    pad(now.getDate(), 2) +
    pad(now.getHours(), 2) +
    pad(now.getMinutes(), 2) +
    pad(now.getSeconds(), 2) +
    pad(now.getMilliseconds(), 3); // 17자
  const random = pad(Math.floor(Math.random() * 10_000_000_000), 10); // 10자
  return `${timestamp}${SYSTEM_CODE}${random}`;
}

let _currentGuid = '';

export function setCurrentGuid(guid: string): void {
  _currentGuid = guid;
}

export function getCurrentGuid(): string {
  return _currentGuid;
}

File: service/be-report/app/middleware/guid.py
import random
from datetime import datetime

SYSTEM_CODE = "LTP"


def generate_guid() -> str:
    now = datetime.now()
    timestamp = now.strftime("%Y%m%d%H%M%S") + f"{now.microsecond // 1000:03d}"
    rand = str(random.randint(0, 9_999_999_999)).zfill(10)
    return f"{timestamp}{SYSTEM_CODE}{rand}"